packages/puppeteer-core/src/bidi/Connection.ts

```diff
@@ -100,8 +100,9 @@ export class BidiConnection
     params: Commands[T]['params'],
     timeout?: number,
   ): Promise<{result: Commands[T]['returnType']}> {
-    assert(!this.#closed, 'Protocol error: Connection closed.');
-
+    if (this.#closed) {
+      return Promise.reject(new Error('Connection closed.'));
```

Comment:
I am thinking if it makes sense to create a dedicated error subclass for ConnectionErrors?

Reply:
I think we should